src/server.rs

```diff
@@ -889,12 +889,14 @@ where
     async fn get_info(&self) -> Result<ServerInfo> {
         let stats = self.stats.read().await.clone();
         let cache_location = self.storage.location();
+        let version = env!("CARGO_PKG_VERSION").to_string();
```

Comment:
Hm, seems that this brings a lot of cascade changes to manage lifetimes if I get you right.
- Define version struct field as ```&'static str'``` on ServerInfo instead of String.
- As ServerInfo struct has _serde::Deserialize_, which has the _'de_ lifetime, complains about it must outlive _'static._ 
However, some of the [_proposals_ ](https://github.com/serde-rs/serde/issues/1712) does not work. So if I try to put a generic one
````rust
pub struct ServerInfo<'a>{
    pub version: &'a str,
}
````

this causes to change different data structs & returns (```Result<ServerInfo<'static>>```, ```Response<'a>```,
```type SccacheResponse<'a> = Message<Response<'a>, Body<Response<'a>>>;```

or ``` type Response<'a> = SccacheResponse<'a>;```` unstable generic associated types...

I think I have misunderstood this or I'm missing something. 

If I'm not completely clueless, I could try and make changes, but I'm afraid I will probably mess something with these lifetimes annotations a few times. 

Sorry for all the noise! Thanks

Reply:
I forgot about serde pains. leave as is then